src/inmanta/agent/handler.py

```diff
@@ -909,7 +909,15 @@ def execute(self, ctx: HandlerContext, resource: resources.Resource, dry_run: Op
                     changes["purged"] = dict(desired=desired.purged, current=True)
 
             for field, values in changes.items():
-                ctx.add_change(field, desired=values["desired"], current=values["current"])
+                for status in ["desired", "current"]:
```

Comment:
Wouldn't it be better to move this logic into the `AttributeStateChange` class? (I might be missing something here)

Reply:
or even in the add_change method, which is another codepath that needs this check.

Reply:
@wouterdb I don't think that just adding it to the `add_change()` method would be sufficient. The way I see it, is the following: Any code path that updates the `self._changes` field of the `HandlerContext` class should be protected against unserializable things (add_change(), add_changes(), update_changes(), etc). All these code paths use the `AttributeStateChange()` class, so adding guard there would ensure that all code paths are safe. Do you agree?